libnd4j/include/ops/declarable/impl/LegacyReduceLongOp.cpp

```diff
@@ -130,6 +131,22 @@ sd::Status LegacyReduceLongOp::validateAndExecute(Context& block) {
   }
 
   manager.synchronize();
+
+
+  if(OpRegistrator::getInstance().traceOps()) {
+    std::vector<const sd::LongType *> *inputShapeBuffers = new std::vector<const sd::LongType *>();
+    for(int i = 0; i < block.width(); i++) {
+      inputShapeBuffers->push_back(block.variable(i)->getNDArray()->shapeInfo());
+    }
+    std::vector<const sd::LongType *> *outputShapeBuffers = new std::vector<const sd::LongType *>();
+    for(int i = 0; i < block.outputWidth(); i++) {
+      outputShapeBuffers->push_back(block.fastpath_out()[i]->shapeInfo());
+    }
+
+    OpExecTrace *opExecTrace = new OpExecTrace(inputShapeBuffers,outputShapeBuffers,this->getOpName());
+    OpRegistrator::getInstance().registerOpExec(opExecTrace);
+  }
+
```

Comment:
This kind of block repeats quite often, is there no way of abstracting that away into a commonly used function?

Reply:
Done.